### **OCEAN - Attrition Risk Formula**
Building a risk formula for attrition based on the OCEAN personality model results.<br>
First import the needed libraries:

In [2]:
import numpy as np

Now we create the 5 variables we want, by creating arrays for each one containing the correlation value discovered in the different reaserch studies. Each study made particular discoveries, with different correlation numbers, but similar enough to discover a clear pattern:

In [20]:
Odata = np.array([-0.02, -0.085, -0.024, -0.05, -0.032])
Cdata = np.array([-0.23, -0.219, -0.257, -0.153, -0.21])
Edata = np.array([-0.09, -0.065, -0.05, -0.072, -0.08])
Adata = np.array([-0.16, -0.138, -0.157, -0.14, -0.136])
Ndata = np.array([0.28, 0.332, 0.328, 0.276, 0.30])

Since we can only use one number for then formula, let us create variables for the mean:

In [21]:
Omean = Odata.mean()
Cmean = Cdata.mean()
Emean = Edata.mean()
Amean = Adata.mean()
Nmean = Ndata.mean()

And for ease of use, let us create an array with the means (we will probably use this later when we build our model:

In [22]:
OCEAN = np.array([Omean, Cmean, Emean, Amean, Nmean])
OCEAN

array([-0.0422, -0.2138, -0.0714, -0.1462,  0.3032])

Our means, in a way, define how much each of the variables affects the risk of attrition, so, with this data, if we had to build a formula right now, it would look like this:

AttRisk = O*-0.0422 + C*-0.2138 + A*-0.1462 + E*-0.0714 + N*0.3032

We can start testing it by creating OCEAN variables using the world averages we do have.

In [207]:
O,C,E,A,N = 0.733941, 0.701998, 0.672329, 0.696807, 0.574399

In [209]:
AttRisk = O*Omean + C*Cmean + A*Amean + E*Emean + N*Nmean

In [210]:
AttRisk 

-0.15677917979999997

Nonetheless, this is not enough, since the results of this formula make no sense. This is because we are not interpreting the result, and to do that, we need to do a few things to it. First off, we need to normalize it, which means using a normalization formula with the minimum and maximum possible results. This would be the new formula with normalization:

In [140]:
AttRisknorm = (((O*Omean + C*Cmean + A*Amean + E*Emean + N*Nmean) - -0.4736) / (0.3032 - -0.4736))

In [141]:
AttRisknorm

0.407853785015448

The result now makes more sense, but 40% risk of turnover for an average person seems very high. Then the question is, what is the what is the average turnover rate in the world. In 2020 it was 10.9%, meaning that an average person then, shold have a 10.9 percent probability of leaving the company within the first year.<br>
That means we have to plug this into our formula, so we come up with something like this:

In [204]:
AttRiskFinal = ((((O*Omean + C*Cmean + A*Amean + E*Emean + N*Nmean) - -0.4736) / (0.3032 - -0.4736))) / 3.74

In [205]:
AttRiskFinal

0.10905181417525348

Now it makes more sense, an average person's risk of attrition, if we want to be as close to reality as possible, is 10.9%. If all the information we collected, and all the averages are actually close to reality, we should now have a very decent risk formula for our needs. But again, it depends on a lot of buts, and here is where we discover one of then bigger, more important truths about data science: our insights are only as good as our data (or in more common terms, "garbage in, garbage out").